include/neighbor/Runtime.h

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2018-2019, Michael P. Howard.
+// This file is released under the Modified BSD License.
+
+// Maintainer: mphoward
+
+#ifndef NEIGHBOR_RUNTIME_H_
+#define NEIGHBOR_RUNTIME_H_
+
+#include <functional>
```

Comment:
We will want to be able to use this in JIT compilation, too, which places additional constraints on not being able to include system headers. See https://github.com/glotzerlab/hoomd-blue/pull/549

Therefore, any includes should be guarded
```C++
#ifdef __CUDACC_RTC__
#define GPURTC
#endif

#if !defined(GPURTC)
#include <functional>
#include <utility>

// any non-kernel code
..
#endif
```

Reply:
I'm not very familiar with NVRTC, so could you clarify--which parts of the runtime API need to be exposed in a file using that? I think `<functional>` is vestigial and can be removed here. `<utility>` is needed for perfect forwarding to the kernel launch (via `std::forward`), and so only needs to be included if `KernelLaunch` is defined.